src/compiler/crystal/tools/context.cr

```diff
@@ -289,9 +292,10 @@ module Crystal
     private def contains_target(node)
       if loc_start = node.location
         loc_end = node.end_location || loc_start
-        # if it is not between, it could be the case that node is the top level Expressions
+        # if it is not between, it could be the case that node is the top level Expressions,
         # in which the (start) location might be in one file and the end location in another.
-        @target_location.between?(loc_start, loc_end) || loc_start.filename != loc_end.filename
+        @target_location.between?(loc_start, loc_end) || (!@in_defs && loc_start.filename != loc_end.filename)
+        # When node generated by macro and macro has yielding block, it also makes `loc_start.filename != loc_end.filename` true.
```

Comment:
It is just ad-hoc solution. The radical problem is `yield` in macro expansion sets its location, but doesn't reset it. Now, I have an idea for it. I'll make a new pull request to solve it and I am going to revert this commit after this pull request is merged.